[CI] MetaDataIndexTemplateServiceTests.testFindTemplatesWithHiddenIndices failing in CI

There have been a [handful of these errors](https://gradle-enterprise.elastic.co/scans/tests?failures.failureClassification=non_verification&list.offset=0&list.size=50&list.sortColumn=startTime&list.sortOrder=desc&search.buildToolType=gradle&search.buildToolType=maven&search.startTimeMax=1579852799999&search.startTimeMin=1578816000000&search.tags=CI&tests.container=org.elasticsearch.action.admin.indices.template.put.MetaDataIndexTemplateServiceTests&tests.sortField=FAILED&tests.test=testFindTemplatesWithHiddenIndices&tests.unstableOnly&trends.section=overview&trends.timeResolution=day&viewer.tzOffset=-480) in CI. Since this is a new test I suspect this might just be an issue with the test.

```
:server:test » org.elasticsearch.action.admin.indices.template.put.MetaDataIndexTemplateServiceTests » testFindTemplatesWithHiddenIndices (0.365s)

Expected: iterable containing ["bar", "global"]
     but: item 0: was "global"
java.lang.AssertionError: 

Expected: iterable containing ["bar", "global"]
     but: item 0: was "global"
Open stacktrace
[2020-01-22T23:53:46,425][INFO ][o.e.a.a.i.t.p.MetaDataIndexTemplateServiceTests] [testFindTemplatesWithHiddenIndices] before test
[2020-01-22T23:53:46,430][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s_0] removing template [one_shard_index_template]
[2020-01-22T23:53:46,430][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s_0] removing template [random-soft-deletes-template]
[2020-01-22T23:53:46,494][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s_0] adding template [foo-1] for index patterns [foo-*]
[2020-01-22T23:53:46,532][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s_0] adding template [foo-2] for index patterns [foo-*]
[2020-01-22T23:53:46,555][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s_0] adding template [bar] for index patterns [bar-*]
[2020-01-22T23:53:46,582][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s_0] adding template [global] for index patterns [*]
[2020-01-22T23:53:46,607][INFO ][o.e.c.m.MetaDataIndexTemplateService] [node_s_0] adding template [sneaky-hidden] for index patterns [sneaky*]
[2020-01-22T23:53:46,630][INFO ][o.e.a.a.i.t.p.MetaDataIndexTemplateServiceTests] [testFindTemplatesWithHiddenIndices] after test
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.action.admin.indices.template.put.MetaDataIndexTemplateServiceTests.testFindTemplatesWithHiddenIndices" -Dtests.seed=5C98A285EE9F1B9F -Dtests.security.manager=true -Dtests.locale=ro -Dtests.timezone=Etc/GMT -Dcompiler.java=13
```

@jaymode do you mind taking a look?

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/template/put/MetaDataIndexTemplateServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/template/put/MetaDataIndexTemplateServiceTests.java
@@ -167,7 +167,7 @@ public class MetaDataIndexTemplateServiceTests extends ESSingleNodeTestCase {
 
         // hidden
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "foo-1234", true).stream()
-            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("foo-2", "foo-1"));
+            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), containsInAnyOrder("foo-2", "foo-1"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "bar-xyz", true).stream()
             .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("bar"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "baz", true), empty());
@@ -176,9 +176,9 @@ public class MetaDataIndexTemplateServiceTests extends ESSingleNodeTestCase {
 
         // not hidden
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "foo-1234", false).stream()
-            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("foo-2", "foo-1", "global"));
+            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), containsInAnyOrder("foo-2", "foo-1", "global"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "bar-xyz", false).stream()
-            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("bar", "global"));
+            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), containsInAnyOrder("bar", "global"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "baz", false).stream()
             .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("global"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "sneaky1", false).stream()
@@ -186,9 +186,9 @@ public class MetaDataIndexTemplateServiceTests extends ESSingleNodeTestCase {
 
         // unknown
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "foo-1234", null).stream()
-            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("foo-2", "foo-1", "global"));
+            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), containsInAnyOrder("foo-2", "foo-1", "global"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "bar-xyz", null).stream()
-            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("bar", "global"));
+            .map(IndexTemplateMetaData::name).collect(Collectors.toList()), containsInAnyOrder("bar", "global"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "baz", null).stream()
             .map(IndexTemplateMetaData::name).collect(Collectors.toList()), contains("global"));
         assertThat(MetaDataIndexTemplateService.findTemplates(state.metaData(), "sneaky1", null).stream()

